src/yarp.c

```diff
@@ -1548,13 +1658,88 @@ yp_if_node_ternary_create(yp_parser_t *parser, yp_node_t *predicate, const yp_to
   return yp_if_node_create(parser, question_mark, predicate, if_statements, (yp_node_t *)else_node, &end_keyword);
 }
 
+// Allocate and initialize a new IntegerNode node.
+static yp_integer_node_t *
+yp_integer_node_create(yp_parser_t *parser, const yp_token_t *token) {
+  assert(token->type == YP_TOKEN_INTEGER);
+  yp_integer_node_t *node = yp_alloc(parser, sizeof(yp_integer_node_t));
+  *node = (yp_integer_node_t) {{ .type = YP_NODE_INTEGER_NODE, .location = YP_LOCATION_TOKEN_VALUE(token) }};
+  return node;
+}
+
+// Allocate and initialize a new RationalNode node.
+static yp_rational_node_t *
+yp_rational_node_create(yp_parser_t *parser, const yp_token_t *token) {
+  assert(token->type == YP_TOKEN_RATIONAL_NUMBER);
+  assert(parser->lex_modes.current->mode == YP_LEX_NUMERIC);
+
+  yp_node_t *numeric_node;
+  yp_token_t numeric_token = {
+    .type = parser->lex_modes.current->as.numeric.type,
+    .start = parser->lex_modes.current->as.numeric.start,
+    .end = parser->lex_modes.current->as.numeric.end
+  };
+  switch (parser->lex_modes.current->as.numeric.type) {
+    case YP_TOKEN_INTEGER: {
+      numeric_node = (yp_node_t *)yp_integer_node_create(parser, &numeric_token);
+      break;
+    }
+    case YP_TOKEN_FLOAT: {
+      numeric_node = (yp_node_t *)yp_float_node_create(parser, &numeric_token);
+      break;
+    }
+    default: {
+      assert(false && "unreachable rational base");
```

Comment:
I _think_ you'll still have to assign a value to `numeric_node` in the default case so it doesn't complain when assertions are turned off that there's no value in this case.